feat(List): Convert list to TypeScript

rebeccaalpert opened the pull request:
Converted to TypeScript.

Fixes https://github.com/patternfly/patternfly-react/issues/1994.

patternfly-build commented:
PatternFly-React preview: https://1946-pr-patternfly-react-patternfly.surge.sh

nicolethoen commented:
@rebeccaalpert can you also add integration tests for this component as outlined on this [README](https://github.com/patternfly/patternfly-react/tree/master/packages/patternfly-4/react-integration)?

dlabaj commented on packages/patternfly-4/react-core/src/components/List/index.ts:
```diff
@@ -0,0 +1,2 @@
+export { List } from './List';
```
Need to convert these to export * with the ts files since Babel only works on a file by file basis.
```suggestion
export * from './List';
export * from './ListItem';
```

rebeccaalpert commented:
Added demo and test. I also addressed @dlabaj's PR feedback. @seanforyou23, as noted on Slack, git mv doesn't seem to net a different outcome, so I'm leaving that as-is for now.
